pkgs/development/ocaml-modules/chacha/default.nix

```diff
@@ -0,0 +1,41 @@
+{ lib
+, buildDunePackage
+, fetchurl
+, ocaml
+
+, alcotest
+, cstruct
+, mirage-crypto
+}:
+
+buildDunePackage rec {
+  pname = "chacha";
+  version = "1.0.0";
+
```

Comment:
`chaca` has support for `ocaml-freestanding`. You can grep `pkgs/development/ocaml-modules` for inspiration if you want to add it (it's not a hard requirement though).